param/__init__.py

```diff
@@ -1497,6 +1501,15 @@ def _validate(self, val):
         if self.rows is not None:
             self._length_bounds_check(self.rows, len(val), 'Row')
 
+    @classmethod
+    def serialize(cls, value):
+        return value.to_dict('records')
+
+    @classmethod
+    def deserialize(cls, value):
+        from pandas import DataFrame as pdDFrame
+        return pdDFrame(value)
```

Comment:
This may not be sufficient to fully reconstruct things since:

a) The column order isn't encoded
b) The index columns can't be reconstructed this way

I wonder if it would be helpful if `deserialize` could optionally accept the schema to help it reconstruct the original data better (it may not help in this particular case though).

Reply:
Not sure what to do about the index but I do know that the JSON schema for the row-oriented data can't be relied on to supply the column order as the columns are declared in JSON objects/Python dictionaries. Python dictionaries *could* encode order information assuming the order of JSON parsing is preserved but I'm not sure this is behavior we should rely on.

Reply:
The JSON spec says not to rely on property ordering https://github.com/json-schema-org/json-schema-spec/issues/571

As that issue suggests we may look at extensions to json-schema such as json-schema-form.

Reply:
There seems to have been discussion about a `propertyOrder` specification but it doesn't seem to have gone anywhere https://github.com/json-schema/json-schema/issues/119